presto-main/src/main/java/com/facebook/presto/operator/MarkDistinctOperator.java

```diff
@@ -134,44 +138,71 @@ public void finish()
     @Override
     public boolean isFinished()
     {
-        return finishing && outputPage == null;
+        return finishing && inputPage == null && unfinishedWork == null;
```

Comment:
Can you add a hasUnfinishedInput method like the in the RowNumber commit? I think it made the code easier to read?